source/PlayState.hx

```diff
@@ -1,29 +1,162 @@
 package;
 
 import flixel.text.FlxText;
+import entities.projectiles.Fireball;
+import entities.collectables.Coin;
+import ui.HUD;
+import flixel.FlxObject;
 import flixel.util.FlxColor;
+import flixel.FlxG;
+import flixel.group.FlxGroup.FlxTypedGroup;
+import entities.player.Hero;
+import entities.terrain.Wall;
 import flixel.FlxState;
 
 class PlayState extends FlxState
 {
-	override public function create()
+	private static var WALL_COUNT(default, never) = 10;
+	private static var WALL_START_X(default, never) = 150;
+	private static var WALL_START_Y(default, never) = 200;
+
+	private static var COIN_COUNT(default, never) = 5;
+	private static var COIN_SPAWN_BORDER(default, never) = 50;
+
+	private static var FIREBALL_COUNT(default, never) = 10;
+	private static var FIREBALL_SPAWN_BORDER(default, never) = 50;
+
+	var hud:HUD;
+	var token:Int = 0;
+
+	private var winMessage:FlxText;
+	private var loseMessage:FlxText;
+
+	private var hero:Hero;
+	private var walls:FlxTypedGroup<Wall>;
+	private var coins:FlxTypedGroup<Coin>;
+	private var fireballs:FlxTypedGroup<Fireball>;
+
+	override public function create():Void
 	{
+		hud = new HUD();
+		add(hud);
+
 		super.create();
-		var myMsg = new FlxText(125, 200, 0);
-		myMsg.text = "Greetings Humans.";
-		myMsg.setFormat("assets/font.ttf", 50, FlxColor.RED, CENTER);
-		myMsg.setBorderStyle(OUTLINE, FlxColor.BLUE, 1);
-		add(myMsg);
 
-		var player:Hero = new Hero();
+		hero = new Hero();
+		hero.x= 240;
+		hero.y= 170;
+		add(hero);
+
+		initializeWalls();
+		initializeFireballs();
+		initializeCoins();
+	}
+
+	private function initializeWalls() {
+		walls = new FlxTypedGroup<Wall>();
+
+		for (i in 0...WALL_COUNT) {
+			var x:Float = WALL_START_X + (i * Wall.WIDTH);
+			var y:Float = WALL_START_Y;
+			var wall:Wall = new Wall(x, y);
+			walls.add(wall);
+		}
+		add(walls);
+	}
+	private function initializeCoins() {
+		coins = new FlxTypedGroup<Coin>();
+
+		for (i in 0...COIN_COUNT)
+			{
+				var x:Float = FlxG.random.int(COIN_SPAWN_BORDER,
+					 FlxG.width - COIN_SPAWN_BORDER);
+				var y:Float = FlxG.random.int(COIN_SPAWN_BORDER,
+					 FlxG.height - COIN_SPAWN_BORDER);
+				var coin = new Coin(x , y);
+				coins.add(coin);
+			}
+		add(coins);
+	}
 
-		add(player);
+	private function initializeFireballs() {
+		fireballs = new FlxTypedGroup<Fireball>();
 
+		for (i in 0...FIREBALL_COUNT) {
+			var x:Float = FlxG.random.int(FIREBALL_SPAWN_BORDER, 
+				FlxG.width - FIREBALL_SPAWN_BORDER);
+			var y:Float = FlxG.random.int(FIREBALL_SPAWN_BORDER, 
+				FlxG.height - FIREBALL_SPAWN_BORDER);
```

Comment:
This fireball spawning logic ensures that all fireballs spawn within the border specified at the top of the class, which lets them fill the middle of the screen. (I expect that this was intentional)

This does have the downside that the player sometimes just immediately gets killed due to a fireball spawning inside of them, at which point they'd need to close and re-open the game to play again. Might be worth considering ways to mitigate the odds of this happening, or just make it easier for the player to restart.